contracts/contracts/MessageProcessor.sol

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {AccQueue} from "./trees/AccQueue.sol";
+import {IMACI} from "./IMACI.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Poll} from "./Poll.sol";
+import {SnarkCommon} from "./crypto/SnarkCommon.sol";
+import {CommonUtilities} from "./Utility.sol";
+import {Verifier} from "./crypto/Verifier.sol";
+import {VkRegistry} from "./VkRegistry.sol";
+
+contract MessageProcessor is Ownable, SnarkCommon, CommonUtilities {
+
+    string constant ERROR_VOTING_PERIOD_PASSED = "ProcessE01";
+    string constant ERROR_VOTING_PERIOD_NOT_PASSED = "ProcessE02";
+    string constant ERROR_NO_MORE_MESSAGES = "ProcessE03";
+    string constant ERROR_STATE_AQ_NOT_MERGED = "ProcessE04";
+    string constant ERROR_MESSAGE_AQ_NOT_MERGED = "ProcessE04";
+    string constant ERROR_INVALID_PROCESS_MESSAGE_PROOF = "ProcessE05";
+    string constant ERROR_VK_NOT_SET = "ProcessE06";
+
+
+    // Whether there are unprocessed messages left
+    bool public processingComplete;
+    // The number of batches processed
+    uint256 public numBatchesProcessed;
+    // The current message batch index. When the coordinator runs
+    // processMessages(), this action relates to messages
+    // currentMessageBatchIndex to currentMessageBatchIndex + messageBatchSize.
+    uint256 public currentMessageBatchIndex;
+   // The commitment to the state and ballot roots
+    uint256 public sbCommitment;
+
+    Verifier public verifier;
+
+    constructor(Verifier _verifier) {
+        verifier = _verifier;
+    }
+
+    modifier votingPeriodOver(Poll _poll) {
```

Comment:
Let's move this to the CommonUtilities contract as an internal function so we can use it everywhere else

Reply:
yeah